marc_to_solr/lib/princeton_marc.rb

```diff
@@ -341,20 +343,213 @@ def oclc_normalize oclc, opts = {prefix: false}
   end
 end
 
+# Cached mapping of ARKs to Bib IDs
+# Retrieves and stores paginated Solr responses containing the ARK's and BibID's
+class CacheMap
+  attr_reader :values
+
+  # Constructor
+  # @param host [String] the host for the Blacklight endpoint
+  # @param path [String] the path for the Blacklight endpoint
+  # @param rows [Integer] the number of rows for each Solr response
+  # @param logger [IO] the logging device
+  def initialize(host:, path: '/catalog.json', rows: 1000000, logger: STDOUT)
+    @host = host
+    @path = path
+    @rows = rows
+    @logger = logger
+    @values = {}
+
+    seed!
+  end
+
+  # Seed the cache
+  # @param page [Integer] the page number at which to start the caching
+  def seed!(page: 1)
+    response = query(page: page)
+    return if response.empty?
+
+    pages = response.fetch('pages')
+
+    cache_page(response)
+
+    if pages.fetch('last_page?') == false
+      seed!(page: page + 1)
+    end
+  end
+
+  private
+
+    # Cache a page
+    # @param page [Hash] Solr response page
+    def cache_page(page)
+      docs = page.fetch('docs')
+      docs.each do |doc|
+        arks = doc.fetch('identifier_ssim', [])
+        bib_ids = doc.fetch('source_metadata_identifier_ssim', [])
+
+        ark = arks.first
+        bib_id = bib_ids.first
+
+        @values[ark] = bib_id
+      end
+    end
+
+    # Query the service using the endpoint
+    # @param [Integer] the page parameter for the query
+    def query(page: 1)
+      begin
+        url = URI::HTTPS.build(host: @host, path: @path, query: 'q=&rows=1000000&page=1&f[identifier_tesim][]=ark')
```

Comment:
The `page` parameter needs to be a variable instead of hard-coded as `1`, right?